src/features/environmentDetails/components/EnvironmentDetails.tsx

```diff
@@ -34,30 +38,67 @@ import artifactList from "../../../utils/helpers/artifact";
 import createLabel from "../../../common/config/labels";
 import { AlertDialog } from "../../../components/Dialog";
 import { useAppDispatch, useAppSelector } from "../../../hooks";
-import { CondaSpecificationPip } from "../../../common/models";
+import {
+  CondaSpecificationPip,
+  Environment,
+  Namespace
+} from "../../../common/models";
 import { useInterval } from "../../../utils/helpers";
+import { showNotification } from "../../notification/notificationSlice";
+import { useScrollRef } from "../../../layouts/PageLayout";
 
-interface IEnvDetails {
-  scrollRef: any;
-  environmentNotification: (notification: any) => void;
-}
 interface IUpdateEnvironmentArgs {
   dependencies: (string | CondaSpecificationPip)[];
   channels: string[];
 }
 
 const INTERVAL_REFRESHING = 5000;
 
-export const EnvironmentDetails = ({
-  scrollRef,
-  environmentNotification
-}: IEnvDetails) => {
+export const EnvironmentDetails = () => {
   const dispatch = useAppDispatch();
+
+  // Url routing params
+  // If user loads the app at /<namespace_name>/<environment_name>
+  // This will put the app in the correct state
+  const { namespaceName, environmentName } = useParams<{
+    namespaceName: string;
+    environmentName: string;
+  }>();
+  const namespaces: Namespace[] = useAppSelector(
+    state => state.namespaces.data
+  );
+  const namespace = namespaces.find(({ name }) => name === namespaceName);
+  const foundNamespace = Boolean(namespace);
+  const environments: Environment[] = useAppSelector(
+    state => state.environments.data
+  );
+  const environment = environments.find(
+    environment =>
+      environment.namespace.name === namespaceName &&
+      environment.name === environmentName
+  );
+  const foundEnvironment = Boolean(environment);
+  useEffect(() => {
+    if (namespace && environment) {
+      dispatch(
+        environmentOpened({
+          environment,
+          canUpdate: namespace.canUpdate
+        })
+      );
+      dispatch(modeChanged(EnvironmentDetailsModes.READ));
+      dispatch(toggleNewEnvironmentView(false));
+    }
+  }, [namespaceName, environmentName, foundNamespace, foundEnvironment]);
```

Comment:
It's probably not clear what I'm doing here. I'll need to add a comment or something.

I want this React hook to run whenever the namespace and environment name changes, but I also need it to re-run when an environment or namespace have been found. 

Previously this component (EnvironmentDetails) was only rendered once the environment details had been fetched from the server (see the old version of PageLayout in the files changed). 

Now the component renders before the data has been fetched from the server. So this hook will run once but nothing will happen because namespace and environment are undefined. Then once those are fetched, the hook will run again.

Most of the changes to this component (including passing the Redux toolkit skip token, checking for current build id, and the check for selected environment) have to do with it now being rendered before all the data has arrived.

Reply:
I think I overcomplicated this. I'm going to push a commit that should simplify this.